src/common/login.js

```diff
@@ -1,54 +1,52 @@
-//             date_first_contact ? `&date_first_contact=${date_first_contact}` : ''
-//         }`
-//         const affiliate_tracking = Cookies.getJSON('affiliate_tracking')
-//         const utm_data = TrafficSource.getData()
-//         const utm_source = TrafficSource.getSource(utm_data)
-//         const utm_source_link = utm_source ? `&utm_source=${utm_source}` : ''
-//         const utm_medium_link = utm_data.utm_medium ? `&utm_medium=${utm_data.utm_medium}` : ''
-//         const utm_campaign_link = utm_data.utm_campaign
-//             ? `&utm_campaign=${utm_data.utm_campaign}`
-//             : ''
-//         const affiliate_token_link = affiliate_tracking
-//             ? `&affiliate_token=${affiliate_tracking}`
-//             : ''
-//         const deriv_app_app_id = 16929
+    const loginUrl = () => {
+        const server_url = localStorage.getItem('config.server_url')
+        const language = localStorage.getItem('i18n')?.replace('-', '_')
+        const signup_device = LocalStore.get('signup_device') || (isMobile() ? 'mobile' : 'desktop')
+        const date_first_contact = LocalStore.get('date_first_contact')
+        const marketing_queries = `&signup_device=${signup_device}${
+            date_first_contact ? `&date_first_contact=${date_first_contact}` : ''
+        }`
+        const affiliate_tracking = Cookies.getJSON('affiliate_tracking')
+        const utm_data = TrafficSource.getData()
+        const utm_source = TrafficSource.getSource(utm_data)
+        const utm_source_link = utm_source ? `&utm_source=${utm_source}` : ''
+        const utm_medium_link = utm_data.utm_medium ? `&utm_medium=${utm_data.utm_medium}` : ''
+        const utm_campaign_link = utm_data.utm_campaign
+            ? `&utm_campaign=${utm_data.utm_campaign}`
+            : ''
+        const affiliate_token_link = affiliate_tracking
+            ? `&affiliate_token=${affiliate_tracking}`
+            : ''
 
-//         return server_url && /qa/.test(server_url)
-//             ? `https://${server_url}/oauth2/authorize?app_id=${getAppId()}&l=${language}${marketing_queries}&brand=${app_name.toLowerCase()}${affiliate_token_link}${utm_source_link}${utm_medium_link}${utm_campaign_link}`
-//             : `https://oauth.derivcrypto.com/oauth2/authorize?app_id=${deriv_app_app_id}&l=${language}${marketing_queries}&brand=${app_name.toLowerCase()}${affiliate_token_link}${utm_source_link}${utm_medium_link}${utm_campaign_link}`
-//     }
+        return server_url && /qa/.test(server_url)
+            ? `https://${server_url}/oauth2/authorize?app_id=${getAppId()}&l=${language}${marketing_queries}&brand=${app_name.toLowerCase()}${affiliate_token_link}${utm_source_link}${utm_medium_link}${utm_campaign_link}`
+            : `https://oauth.deriv.com/oauth2/authorize?app_id=${getAppId()}&l=${language}${marketing_queries}&brand=${app_name.toLowerCase()}${affiliate_token_link}${utm_source_link}${utm_medium_link}${utm_campaign_link}`
```

Comment:
my bad, please change the app id to `1411`, can put it in a constant as deriv_crypto_app_id